From 4b6b37b38a26d68be29531fe8df291ee13604450 Mon Sep 17 00:00:00 2001
From: Anton Fedchin <afedchin@ruswizards.com>
Date: Thu, 9 Jan 2014 18:54:00 +0400
Subject: [PATCH] h264: expose h264 frame packing as stereo_mode metadata

---
 libavcodec/h264.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavcodec/h264.c b/libavcodec/h264.c
index 4fb97e2..ffb7aab 100644
--- a/libavcodec/h264.c
+++ b/libavcodec/h264.c
@@ -5069,6 +5069,7 @@ static int decode_frame(AVCodecContext *avctx, void *data,
             if (ret < 0)
                 return ret;
             *got_frame = 1;
+            av_dict_set(&pict->metadata, "stereo_mode", ff_h264_sei_stereo_mode(h), 0);
         }
 
         return buf_index;
@@ -5127,6 +5128,7 @@ not_extra:
             if (ret < 0)
                 return ret;
             *got_frame = 1;
+            av_dict_set(&pict->metadata, "stereo_mode", ff_h264_sei_stereo_mode(h), 0);
             if (CONFIG_MPEGVIDEO) {
                 ff_print_debug_info2(h->avctx, h->next_output_pic, pict, h->er.mbskip_table,
                                     &h->low_delay,
-- 
1.7.10.msysgit.1

